src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -898,10 +952,10 @@ class TFBertForPreTrainingOutput(ModelOutput):
     BERT_START_DOCSTRING,
 )
 class TFBertModel(TFBertPreTrainedModel):
-    def __init__(self, config, *inputs, **kwargs):
-        super().__init__(config, *inputs, **kwargs)
+    def __init__(self, config: BertConfig, *inputs, **kwargs):
+        super().__init__(config=config, *inputs, **kwargs)
 
-        self.bert = TFBertMainLayer(config, name="bert")
+        self.bert = TFBertMainLayer(config=config, name="bert")
 
```

Comment:
I think this example is exactly why I dislike this change. Before the change, we understood that `TFBertMainLayer` took one parameter, the configuration, and other *named* parameters to configure it further (for example TF-specific arguments like `name`, `training`, others). It's very clear that to initialize it we're only going to need the configuration.

Now, with this change, I think differently for two reasons:
- It looks like the `TFBertMainLayer` has no required arguments. The configuration was passed to it, but it doesn't seem necessary to me. I would even argue it's now on the same level as the `name` parameter, which is not true.
- That means that if I want to configure it, then I'll need to go and take a look at the signature to understand what arguments I may be unaware of that would change the `TFBertMainLayer`'s initialization/behavior.

While beforehand the signature was very clear, now it isn't.